lib/video/screenRecording/desktop/firefoxrecorder.js

```diff
@@ -0,0 +1,161 @@
+'use strict';
+
+const log = require('intel').getLogger('browsertime.video');
+const os = require('os');
+const util = require('util');
+const fs = require('fs');
+const path = require('path');
+const delay = ms => new Promise(res => setTimeout(res, ms));
+const execa = require('execa');
+const del = require('del');
+
+const unlink = util.promisify(fs.unlink);
+
+async function createTempFile() {
+  const mkdtemp = util.promisify(fs.mkdtemp);
+  const dir = await mkdtemp(path.join(os.tmpdir(), 'browsertime-'));
+  return path.join(dir, 'winrecorder.mp4');
+}
+
+async function enableWindowRecorder(enable, driver) {
+  if (enable) {
+    enable = '1';
+  } else {
+    enable = '0';
+  }
+  const oldContext = driver.getContext();
+  await driver.setContext('chrome');
+  const script = 'windowUtils.setCompositionRecording(' + enable + ');';
+  await driver.executeScript(script);
+  return driver.setContext(oldContext);
+}
+
+function findRecordingDirectory() {
+  let closest_mtime = 0;
+  let directory = undefined;
+
+  fs.readdirSync('.').forEach(file => {
+    if (file.startsWith('windowrecording-')) {
+      let mtime = fs.statSync(file).mtime;
+      if (mtime > closest_mtime) {
+        closest_mtime = mtime;
+        directory = file;
+      }
+    }
+  });
+
+  log.debug('Using window recording directory: ' + directory);
+  return directory;
+}
+
+async function pollDirectory(directory) {
+  // We need to spin wait here until the browser is finished
```

Comment:
Add a `// TODO: ...` saying the bug that will make the actual API in Firefox useful (synchronous).